discovery-infra/oc_utils.py

```diff
@@ -35,57 +36,130 @@ def extend_parser_with_oc_arguments(parser):
     )
 
 
-class OC(object):
+class OpenShiftConfiguration(Configuration):
 
-    KUBE_CONFIG_PATH = os.environ.get('KUBECONFIG')
-    OC_DOMAIN = 'redhat.com'
+    def __init__(self):
+        Configuration.__init__(self)
+        self.__verify_ssl = False
 
-    def __init__(self, **kwargs):
-        self.config = None
-        self.load_kube_config(**kwargs)
-        self.client = ApiClient(self.config)
+    @property
+    def token(self):
+        return self.api_key['authorization']
 
-    def load_kube_config(self, token=None, server=None, verify_ssl=False):
-        kube_config.load_kube_config(self.KUBE_CONFIG_PATH)
-        self.config = kube_config.Configuration()
-        if token:
-            self.config.api_key['authorization'] = f'Bearer {token}'
-        if server:
-            self.config.host = server
+    @token.setter
+    def token(self, token):
+        if not token.startswith('Bearer '):
+            token = f'Bearer {token}'
 
-        self.config.verify_ssl = verify_ssl
-        if not self.config.verify_ssl:
+        self.api_key['authorization'] = token
+
+    @property
+    def server(self):
+        return self.host
+
+    @server.setter
+    def server(self, server):
+        self.host = server
+
+    @property
+    def verify_ssl(self):
+        return self.__verify_ssl
+
+    @verify_ssl.setter
+    def verify_ssl(self, verify_ssl):
+        if not verify_ssl:
             urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
 
-        self.validate_conn_params()
-
-    def validate_conn_params(self):
-        if self.OC_DOMAIN not in self.config.host:
-            raise ValueError('oc host is not part of the domain')
-        elif not self.config.auth_settings()['BearerToken']['value']:
-            raise ValueError('oc missing authentication token')
-
-    def get_namespaced_service_routes_list(self, namespace, service):
-        return self.client.call_api(
-            f'/apis/route.openshift.io/v1/namespaces/{namespace}/routes',
-            method='GET',
-            query_params=[('fieldSelector', f'spec.to.name={service}')],
-            response_type='V1ResourceQuotaList',
-            auth_settings=self.config.auth_settings(),
-            _return_http_data_only=True
+        self.__verify_ssl = verify_ssl
+
+
+class OpenShiftAPIClient(ApiClient):
+
+    def call_api(
+            self,
```

Comment:
sure all can be kwargs, they are not interesting params as you use only auth_settings

Reply:
I can't relay on kwargs, this function must get args (so it can be also used by kubernetes library as a replacment for APIClient). And I need to be sure I catch this argument :(